### PR TITLE
Implement bytea HEX encode/decode

### PR DESCRIPTION
**Pull request by davecramer**
this is from upstream commit a2a8c7a662ec96537b6d1faba0770c516b921911
The primary reason for back patching is to allow pgadmin4 to work
Unfortunately there are a few hacks to make this work without Enum Config GUCs
mostly around setting the GUC as the target is an integer but the guc value is a string

### CODE REVIEW

File: src/backend/utils/misc/postgresql.conf.sample [hunk 1]
```diff
@@ -438,7 +438,7 @@ log_autostats=off	# print additional autostats information
 #statement_timeout = 0			# 0 is disabled
 #idle_session_gang_timeout = 18s 	# in milliseconds, 0 is disabled
 #vacuum_freeze_min_age = 100000000
-
+#bytea_output='escape'
```
Comment (hlinnaka) [inline src/backend/utils/misc/postgresql.conf.sample]: Keep a newline between this and the "- Local and Formatting -" comment below

File: src/include/utils/guc.h [hunk 2]
```diff
@@ -113,6 +113,18 @@ extern bool ParseConfigFile(const char *config_file, const char *calling_file,
 							struct name_value_pair **tail_p);
 extern void free_name_value_list(struct name_value_pair * list);
 
+/*
+ * The possible values of an enum variable are specified by an array of
+ * name-value pairs.  The "hidden" flag means the value is accepted but
+ * won't be displayed when guc.c is asked for a list of acceptable values.
+ */
+struct config_enum_entry
```
Comment (hlinnaka) [inline src/include/utils/guc.h]: This seems to be unused.

Comment (davecramer) [inline src/include/utils/guc.h]: Thx, I'll clean it up, left over crud from blindly applying the upstream patch

**Comment by danielgustafsson** [comment]
Pushed commit 894a8d307d397286b21bfc375ea716abd0ec00a0 to master which fixes a compiler warning introduced by this PR.

**Pull request merged**